math/cmath/include/algebra/math/impl/cmath_getter.hpp

```diff
@@ -193,4 +193,113 @@ struct block_getter {
+  ALGEBRA_HOST_DEVICE inline scalar_t operator()(
+      const array_t<array_t<scalar_t, N>, N> &m) const {
+    return determinant_getter_helper<N>()(m);
+  }
+
+  template <size_type N, typename Enable = void>
+  struct determinant_getter_helper;
+
+  template <size_type N>
+  struct determinant_getter_helper<N, typename std::enable_if_t<N == 1>> {
+    template <class input_matrix_type>
+    ALGEBRA_HOST_DEVICE inline scalar_t operator()(
+        const input_matrix_type &m) const {
+      return m[0][0];
+    }
+  };
+
+  template <size_type N>
+  struct determinant_getter_helper<N, typename std::enable_if_t<N != 1>> {
+
+    template <class input_matrix_type>
+    ALGEBRA_HOST_DEVICE inline scalar_t operator()(
+        const input_matrix_type &m) const {
+
+      scalar_t D = 0;
+
+      // To store cofactors
+      array_t<array_t<scalar_t, N>, N> temp;
+
+      // To store sign multiplier
+      int sign = 1;
+
+      // Iterate for each element of first row
+      for (size_type col = 0; col < N; col++) {
+        // Getting Cofactor of A[0][f]
+        this->get_cofactor(m, temp, size_type(0), col);
+        D += sign * m[col][0] * determinant_getter_helper<N - 1>()(temp);
+
+        // terms are to be added with alternate sign
+        sign = -sign;
+      }
+
+      return D;
+    }
+
+    template <class input_matrix_type>
+    ALGEBRA_HOST_DEVICE inline void get_cofactor(
+        const input_matrix_type &m, array_t<array_t<scalar_t, N>, N> &temp,
+        size_type p, size_type q) const {
+
+      size_type i = 0, j = 0;
+
+      // Looping for each element of the matrix
+      for (size_type row = 0; row < N; row++) {
+        for (size_type col = 0; col < N; col++) {
+          //  Copying into temporary matrix only those element
+          //  which are not in given row and column
+          if (row != p && col != q) {
+            temp[j++][i] = m[col][row];
+
+            // Row is filled, so increase row index and
+            // reset col index
+            if (j == N - 1) {
+              j = 0;
+              i++;
+            }
+          }
+        }
+      }
+    }
+  };
+
+  /*
+  template <size_type N>
+  ALGEBRA_HOST_DEVICE inline array_t<array_t<scalar_t, N>, N> adjoint(
+      const array_t<array_t<scalar_t, N>, N> &m) const {
+    return adjoint_getter<N>()(m);
+  }
+
+  template <size_type N, typename Enable = void>
+  struct adjoint_getter;
+
+  template <size_type N>
+  struct adjoint_getter<N, typename std::enable_if_t<N == 1>> {
+    template <class input_matrix_type>
+    ALGEBRA_HOST_DEVICE inline scalar_t operator()(
+        const input_matrix_type &m) const {
+      return array_t<array_t<scalar_t, N>, N>{{1}};
+    }
+  };
+
+  template <size_type N>
+  struct adjoint_getter<N, typename std::enable_if_t<N != 1>> {
+
+    template <class input_matrix_type>
+    ALGEBRA_HOST_DEVICE inline scalar_t operator()(
+        const input_matrix_type &m) const {
+      return m;
+    }
+  };
+  */
```

Comment:
I see that you moved these functions into a different place after all. Let's not forget about removing this code if that other place will be the final place for these.